server/src/test/java/org/elasticsearch/cluster/settings/ClusterSettingsIT.java

```diff
@@ -380,4 +381,23 @@ public void testLoggerLevelUpdate() {
         }
     }
 
+    public void testUserMetadata() {
+        String key = "cluster.metadata." + randomAlphaOfLengthBetween(5, 20);
+        String value = randomRealisticUnicodeOfCodepointLengthBetween(5, 50);
+
+        // Make sure we have both key and value in case this fails
+        logger.info("Attempting to store [{}]: [{}]", key, value);
+
+        final Settings settings = Settings.builder().put(key, value).build();
+        if (randomBoolean()) {
+            client().admin().cluster().prepareUpdateSettings().setPersistentSettings(settings).execute().actionGet();
+            ClusterStateResponse state = client().admin().cluster().prepareState().execute().actionGet();
+            assertEquals(value, state.getState().getMetaData().persistentSettings().get(key));
```

Comment:
can you also add a tests that updates a setting

Reply:
Will do.